eth_utils/abi.py

```diff
@@ -142,7 +142,7 @@ def collapse_if_tuple(abi: Union[ABIComponent, str]) -> str:
     if isinstance(abi, str):
         return abi
 
-    element_type = abi["type"]
+    element_type = abi.get("type")
```

Comment:
This adds protection so we can raise a `TypeError` rather than it raising a `KeyError`.